### PR TITLE
Zoom Effect

### PR DESCRIPTION
**Pull request by schmamps**
Significant improvement to zoom, with animations.

Double click to zoom, click again to zoom out.

### CODE REVIEW

File: site.js [hunk 1]
```diff
@@ -8,21 +22,24 @@
  * @param {any} boxWidth
  * @param {any} boxHeight
  */
-function fitDimsToBounds(imgWidth, imgHeight, boxWidth, boxHeight) {
+function fitDimsToBounds(imgWidth, imgHeight, boxWidth, boxHeight, constrain) {
   var aspect = imgWidth / imgHeight;
+  var sizeMethod = constrain ? Math.min : Math.max;
```
Comment (coreyadkins) [inline site.js:27]: Nice application of the conditional operator. I like it.

File: site.js [hunk 2]
```diff
@@ -8,21 +22,24 @@
...
-    dims.height = Math.round(dims.width / aspect);
+    dims.height = dims.width / aspect;
   } else {
     // fit to height
-    dims.width = Math.round(dims.height * aspect);
+    dims.width = dims.height * aspect;
   }
 
+  dims.width = Math.round(dims.width);
+  dims.height = Math.round(dims.height);
```
Comment (coreyadkins) [inline site.js:42]: Was there a reason you separated out these two steps (the operation and the math.round)? Seems like the previous code was a bit more compact.

File: site.js [hunk 3]
```diff
@@ -42,20 +65,73 @@ function placePhoto(url) {
...
+
+/**
+ * Blow up the image
+ */
+function zoomImage() {
+  var $ref = $('.photo img');
+
+  if ($ref.attr('src')) {
+    var speed = 'slow';
+    var twoEm = $('header').height();
```
Comment (coreyadkins) [inline site.js:103]: This line is a bit confusing to me. Are you using the header height to determine the size of two ems? If so why is it not `var twoEm = 2em;`? Or if it's trying to get the height of the header, a variable name like ``headerHeight` would make more sense.

File: site.js (hunk 2 above)
Comment (schmamps) [inline site.js:42]: I felt it clarified what would come out. 

Looking at it now, there's also the chance that it's more pixel-accurate to save it for the last. Of course the browser would do just fine with scaling anyway, but…

File: site.js (hunk 3 above)
Comment (schmamps) [inline site.js:103]: There's no way to get the size of an 'em' in JavaScript or even CSS. You just have to see what the browser does.

Comment (schmamps) [inline site.js:103]: And really it's not so much about the header as it is the fact that it's a way to get the size of 2em. It's the subtraction we need to frame the element for most, but not all of the screen.

File: site.js (hunk 2 above)
Comment (coreyadkins) [inline site.js:42]: It doesn't seem to make much of a difference for clarity to me, but if it affects accuracy, then we definitely should leave it. I have no problem with it, was just wondering if there was a reason for the change.

File: site.js (hunk 3 above)
Comment (coreyadkins) [inline site.js:103]: Blah, good point, I forgot that ems are relative.

**Comment by coreyadkins** [comment]
:shipit:

**Pull request merged**